src/app/src/components/NavBar.js

```diff
@@ -0,0 +1,122 @@
+};
+
+export default function NavBar() {
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+    const location = useLocation();
+
+    let variant = NAVBAR_VARIANTS.SUBMISSION;
+    if (location.pathname.startsWith('/draw')) {
+        variant = NAVBAR_VARIANTS.DRAW;
+    }
+
+    // placeholders
+    const utilities = ['Raleigh City of', 'Azavea Test Utility'];
+    const selectedUtility =
+        useSelector(state => state.auth.selectedUtility) || utilities[0];
+
+    const utilitySelectOrLabel =
+        variant === NAVBAR_VARIANTS.SUBMISSION && utilities.length > 1 ? (
+            <Select
+                variant='filled'
+                h='40px'
+                w='250px'
+                value={selectedUtility}
+                onChange={e => {
+                    dispatch(setSelectedUtility(e.target.value));
+                }}
+                _focus={{ background: 'white' }}
+            >
+                {utilities.map(u => {
+                    return (
+                        <option key={u} value={u}>
+                            {u}
+                        </option>
+                    );
+                })}
+            </Select>
+        ) : (
+            <Text textStyle='selectedUtility'>{selectedUtility}</Text>
+        );
+
+    const maybeSettingsButton =
+        variant === NAVBAR_VARIANTS.SUBMISSION ? (
+            <IconButton
+                icon={<Icon as={CogIcon} />}
+                aria-label='Settings'
+                onClick={() => {
+                    navigate('/settings');
+                }}
+            />
+        ) : null;
+
+    const exitButton =
+        variant === NAVBAR_VARIANTS.SUBMISSION ? (
+            <Button
+                onClick={() => {
+                    apiClient.post(API_URLS.LOGOUT, {}).then(() => {
+                        dispatch(logout());
+                        navigate('/login');
+                    });
+                }}
+                rightIcon={<Icon as={LogoutIcon} />}
+            >
+                Log out
+            </Button>
+        ) : (
+            <Button
+                onClick={() => {
+                    navigate('/submissions');
+                }}
+                rightIcon={<Icon as={ArrowLeftIcon} />}
+            >
+                Save and back
+            </Button>
+        );
+
+    return (
+        <SimpleGrid
+            columns={3}
+            paddingLeft={10}
+            paddingRight={10}
+            w='100%'
+            h={NAVBAR_HEIGHT}
+            bg='gray.700'
+        >
+            <Flex align='center'>{utilitySelectOrLabel}</Flex>
+
+            <Flex align='center' justify='center'>
+                <Heading size='md' color='white' ml={6}>
+                    Boundary Sync
+                </Heading>
+            </Flex>
+
+            <Flex align='center' gap={4}>
+                <Spacer />
+                {maybeSettingsButton}
+                {exitButton}
```

Comment:
Can these (and `utilitySelectOrLabel`) be decomposed into their own components in this file?

It would make the NavBar component easier to read. We will have to call some hooks twice (`useDispatch`, `useNavigation`), but hopefully React optimizes that under the hood enough to not matter.

I'm thinking something like this:

```jsx
<Flex align='center' gap={4}>
    <Spacer />
    <SettingsButton variant={variant} />
    <ExitButton variant={variant} />
</Flex>
```

Reply:
I like that a lot. What do you think about including `dispatch` as a prop? d21cac6

Reply:
My preference is for the hook because it reduces the dependecy between components.

Reply:
Will do.